contracts/NonfungiblePositionManager.sol

```diff
@@ -153,14 +153,7 @@ contract NonfungiblePositionManager is BaseLiquidityManagement, INonfungiblePosi
         require(params.liquidityDelta != 0, "Must decrease liquidity");
         Position storage position = positions[params.tokenId];
 
-        (uint160 sqrtPriceX96,,,) = PoolStateLibrary.getSlot0(poolManager, position.range.key.toId());
-        (uint256 amount0, uint256 amount1) = LiquidityAmounts.getAmountsForLiquidity(
-            sqrtPriceX96,
-            TickMath.getSqrtRatioAtTick(position.range.tickLower),
-            TickMath.getSqrtRatioAtTick(position.range.tickUpper),
-            params.liquidityDelta
-        );
-        BaseLiquidityManagement.modifyLiquidity(
+        (BalanceDelta liquidityDelta, BalanceDelta feeDelta) = BaseLiquidityManagement.modifyLiquidity(
```

Comment:
by returning fee delta, decreasing liquidity does not need to calculate "principal" liquidity

saves about 4.5k gas